### PR TITLE
fix bug; will not tag job_tag_finished if one of the tasks fail

### PR DESCRIPTION
**Pull request by felix5572**
fix bug; will not tag job_tag_finished if one of the tasks fail

### CODE REVIEW

**Comment by felix5572** [comment]
@Cloudac7  have a look?

**Review by njzjz — APPROVED** [review]

File: dpdispatcher/machine.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 """
 
 script_env_template="""
+export FLAG_TASK_FAIL=0
```
Comment (njzjz) [inline dpdispatcher/machine.py:18]: I think we don't need `export` here, as we only use it in this script.
```suggestion
FLAG_TASK_FAIL=0
```

**Review by njzjz — CHANGES_REQUESTED** [review]

File: dpdispatcher/machine.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
...
 export REMOTE_ROOT={remote_root}
```
Comment (njzjz) [inline dpdispatcher/machine.py:18]:
```suggestion
REMOTE_ROOT={remote_root}
```

**Review by njzjz — CHANGES_REQUESTED** [review]

**Comment by felix5572** [comment]
@njzjz  I found that the variable in parent shell cannot be accessed by child shell. 
So I change the implementation, we use a new file to record whether the task fails.

**Pull request merged**